crates/top/re_sdk/src/lenses/builder.rs

```diff
@@ -0,0 +1,335 @@
+//! Builder API for constructing lenses.
+
+use re_chunk::{ComponentIdentifier, EntityPath, TimelineName};
+use re_log_types::{EntityPathFilter, TimeType};
+use re_types::ComponentDescriptor;
+
+use super::LensError;
+use super::Op;
+use super::ast;
+use super::ast::OneToMany;
+use super::ast::OneToOne;
+use super::ast::Static;
+
+/// Builder for lenses with support for multiple output modes.
+pub struct LensBuilder {
```

Comment:
Usually a good idea to mark builders as `must_use`

```suggestion
#[must_use]
pub struct LensBuilder {
```